### PR TITLE
Data Feeds - Starknet update

### PR DESCRIPTION
**Pull request by khadni**

### CODE REVIEW

File: src/content/data-feeds/starknet/tutorials/snfoundry/consumer-contract.mdx [hunk 1]
```diff
@@ -0,0 +1,189 @@
...
+1. Navigate to the `aggregator_consumer` directory:
+
+   ```bash
+   cd chainlink-starknet/examples/contracts/aggregator_consumer/
+   ```
+
+After you prepare the requirements, check to make sure the required tools are configured correctly by running the tests:
+
+    ```bash
+    snforge test
```
Comment (chris-de-leon-cll) [inline src/content/data-feeds/starknet/tutorials/snfoundry/consumer-contract.mdx:47]: `make test`

File: src/content/data-feeds/starknet/tutorials/snfoundry/sn-devnet-rs.mdx [hunk 2]
```diff
@@ -0,0 +1,233 @@
...
+
+   ```bash
+   [sncast.default]
+   url = "http://127.0.0.1:5050/rpc"
+   ```
+
+After you prepare the requirements, check to make sure the required tools are configured correctly by running the tests:
+
+    ```bash
+    snforge test
```
Comment (chris-de-leon-cll) [inline src/content/data-feeds/starknet/tutorials/snfoundry/sn-devnet-rs.mdx:52]: `make test`

File: src/content/data-feeds/starknet/index.mdx [hunk 3]
```diff
@@ -1,167 +1,84 @@
...
+Starknet is a permissionless decentralized ZK-Rollup operating as an L2 network over Ethereum. Unlike other Ethereum L2 networks, Starknet is not EVM-compatible and uses [Cairo](https://docs.cairo-lang.org/index.html) as its smart contract language. Chainlink Data Feeds are available on Starknet Sepolia as Cairo smart contracts.
```
Comment (dwightjl) [inline src/content/data-feeds/starknet/index.mdx:17]:
```suggestion
Starknet is a permissionless decentralized Zk-rollup operating as an L2 network over Ethereum. Unlike other Ethereum L2 networks, Starknet is not EVM-compatible and uses [Cairo](https://docs.cairo-lang.org/index.html) as its smart contract language. Chainlink Data Feeds are available on Starknet Sepolia as Cairo smart contracts.
```

Aligning with the style from https://chain.link/education-hub/zero-knowledge-rollup

File: src/content/data-feeds/starknet/index.mdx [hunk 4]
```diff
@@ -1,167 +1,84 @@
...
+Experiment with Chainlink Data Feeds on Starknet using [Starknet Devnet RS](https://github.com/0xSpaceShard/starknet-devnet-rs), a local Docker-based testnet environment implemented using Rust.
+The [Starknet Devnet RS guide](/data-feeds/starknet/tutorials/snfoundry/sn-devnet-rs) provides code examples and scripts that enable you to compile, declare, and deploy your own aggregator and consumer contracts. You can use a set of pre-funded accounts to interact with Chainlink Data Feeds on Starknet without deploying to a live network.
 
-  1. In your clone of the [chainlink-starknet](https://github.com/smartcontractkit/chainlink-starknet) repository, change directories to the proxy consumer example:
+## Getting Started (Starkli CLI)
 
-     ```shell
-     cd ./chainlink-starknet/examples/contracts/proxy-consumer/
-     ```
+[Starkli](https://github.com/xJonathanLEI/starkli) is a Starknet standalone CLI similar to [cairo-lang](https://github.com/starkware-libs/cairo-lang) but written in Rust. It enables interaction with a Starknet network without the need for extra tools or dependencies.
```
Comment (dwightjl) [inline src/content/data-feeds/starknet/index.mdx:34]:
```suggestion
[Starkli](https://github.com/xJonathanLEI/starkli) is a Starknet standalone CLI similar to [cairo-lang](https://github.com/starkware-libs/cairo-lang), but it is written in Rust. It enables interaction with a Starknet network without the need for extra tools or dependencies.
```

File: src/content/data-feeds/starknet/index.mdx [hunk 5]
```diff
@@ -1,167 +1,84 @@
...
 
-  1. Run `yarn install` to install the required packages including [StarkNet.js](https://www.starknetjs.com/), [HardHat](https://hardhat.org/), and the [StarkNet Hardhat Plugin](https://shard-labs.github.io/starknet-hardhat-plugin/).
+In this example, you'll utilize the `starkli call` command to read data from a Chainlink Price Feed on Starknet Sepolia. This command requires the proxy aggregator address, the function selector, and an RPC endpoint for Starknet Sepolia.
```
Comment (dwightjl) [inline src/content/data-feeds/starknet/index.mdx:36]:
```suggestion
In this example, you will use the `starkli call` command to read data from a Chainlink Price Feed on Starknet Sepolia. This command requires the proxy aggregator address, the function selector, and an RPC endpoint for Starknet Sepolia.
```

File: src/content/data-feeds/starknet/tutorials/snfoundry/read-data.mdx [hunk 6]
```diff
@@ -0,0 +1,54 @@
...
+    "Deploy and interact with a Consumer Contract": "/data-feeds/starknet/tutorials/snfoundry/consumer-contract",
+    "Experiment with Starknet Devnet RS": "/data-feeds/starknet/tutorials/snfoundry/sn-devnet-rs",
+  }
+metadata:
+  description: "Read Data from Chainlink Data Feeds on StarkNet using Starknet Foundry."
+---
+
+import { CodeSample, CopyText } from "@components"
+
+In this example, you'll read from a Chainlink price feed on Starknet using the Starknet Foundry toolkit. You'll call the `latest_round_data` function on the ETH / USD aggregator proxy contract to read the latest round of data.
```
Comment (dwightjl) [inline src/content/data-feeds/starknet/tutorials/snfoundry/read-data.mdx:17]:
```suggestion
In this example, you will read from a Chainlink price feed on Starknet using the Starknet Foundry toolkit. You will call the `latest_round_data` function on the ETH / USD aggregator proxy contract to read the latest round of data.
```

Avoid contractions for ESL and translation

File: src/content/data-feeds/starknet/tutorials/snfoundry/read-data.mdx [hunk 7]
```diff
@@ -0,0 +1,54 @@
...
+| `block_num`  | `0xd44d`                              | `54349`                                   | The block number at which the data was recorded on the blockchain                                                |
+| `started_at` | `0x6606b8cc`                          | `1711716556`                              | The Unix timestamp indicating when the data round started                                                        |
+| `updated_at` | `0x6606b8a2`                          | `1711716514`                              | The Unix timestamp indicating when the data was last updated                                                     |
+
+For a complete list of Chainlink Price Feeds available on Starknet, see the [Price Feed Contract Addresses](/data-feeds/price-feeds/addresses?network=starknet) page.
+
+Note: This example utilizes a [Blast API](https://blastapi.io/public-api/starknet) RPC endpoint. You can interact with the network using any other Starknet Sepolia RPC provider, such as [Alchemy](https://www.alchemy.com/starknet) or [Infura](https://www.infura.io/networks/ethereum/starknet).
```
Comment (dwightjl) [inline src/content/data-feeds/starknet/tutorials/snfoundry/read-data.mdx:54]:
```suggestion
Note: This example uses a [Blast API](https://blastapi.io/public-api/starknet) RPC endpoint. You can interact with the network using any other Starknet Sepolia RPC provider, such as [Alchemy](https://www.alchemy.com/starknet) or [Infura](https://www.infura.io/networks/ethereum/starknet).
```

File: src/content/data-feeds/starknet/index.mdx [hunk 8]
```diff
@@ -1,167 +1,84 @@
...
+| `started_at` | `0x000000000000000000000000000000000000000000000000000000006606b8cc` | `1711716556`                              | The Unix timestamp indicating when the data round started                                                        |
+| `updated_at` | `0x000000000000000000000000000000000000000000000000000000006606b8a2` | `1711716514`                              | The Unix timestamp indicating when the data was last updated                                                     |
 
-The command prints the result:
+For a complete list of Chainlink Price Feeds available on Starknet, see the [Price Feed Contract Addresses](/data-feeds/price-feeds/addresses?network=starknet) page.
 
-```text
-0x100000000000000000000000000017c51 6 613174 1673368490 1673368507
-```
+Note: This example utilizes a [Blast API](https://blastapi.io/public-api/starknet) RPC endpoint. You can interact with the network using any other Starknet Sepolia RPC provider, such as [Alchemy](https://www.alchemy.com/starknet) or [Infura](https://www.infura.io/networks/ethereum/starknet).
```
Comment (dwightjl) [inline src/content/data-feeds/starknet/index.mdx:84]:
```suggestion
Note: This example uses a [Blast API](https://blastapi.io/public-api/starknet) RPC endpoint. You can interact with the network using any other Starknet Sepolia RPC provider, such as [Alchemy](https://www.alchemy.com/starknet) or [Infura](https://www.infura.io/networks/ethereum/starknet).
```

File: src/content/data-feeds/starknet/tutorials/snfoundry/consumer-contract.mdx [hunk 9]
```diff
@@ -0,0 +1,189 @@
...
+   git clone https://github.com/smartcontractkit/chainlink-starknet.git
+   ```
+
+1. Navigate to the `aggregator_consumer` directory:
+
+   ```bash
+   cd chainlink-starknet/examples/contracts/aggregator_consumer/
+   ```
+
+After you prepare the requirements, check to make sure the required tools are configured correctly by running the tests:
```
Comment (dwightjl) [inline src/content/data-feeds/starknet/tutorials/snfoundry/consumer-contract.mdx:44]:
```suggestion
After you prepare the requirements, make sure the required tools are configured correctly by running the tests:
```

File: src/content/data-feeds/starknet/tutorials/snfoundry/consumer-contract.mdx [hunk 10]
```diff
@@ -0,0 +1,189 @@
...
+   From the output, note:
+
+   - Your account address. In this example, it is `0x03e7ee3d05d3efae4e493c9733c8c391d4a5cc63d34f95a164c832060fcdd43d`, with an added zero after `0x`.
+   - The `max_fee` value, which is the minimum amount of testnet ETH required to deploy the account.
+
+1. Fund the newly created account with testnet ETH to cover the account deployment, the consumer contract deployment, and the network interaction costs.
+
+   Go to the Blast [Starknet Sepolia ETH Faucet](https://blastapi.io/faucets/starknet-sepolia-eth) and enter your account address to receive testnet ETH. Wait a few seconds for the transaction to complete.
+
+   Alternatively, you can transfer ETH tokens from a Starknet-compatible wallet, such as [Argent](https://www.argent.xyz/) or [Braavos](https://braavos.app/), or utilize the [StarkGate bridge](https://starkgate.starknet.io/) to transfer testnet ETH from Ethereum Sepolia to Starknet Sepolia. If you need testnet ETH on Ethereum Sepolia, you can use the [Chainlink faucet](https://faucets.chain.link/sepolia).
```
Comment (dwightjl) [inline src/content/data-feeds/starknet/tutorials/snfoundry/consumer-contract.mdx:97]:
```suggestion
   Alternatively, you can transfer ETH tokens from a Starknet-compatible wallet, such as [Argent](https://www.argent.xyz/) or [Braavos](https://braavos.app/), or use the [StarkGate bridge](https://starkgate.starknet.io/) to transfer testnet ETH from Ethereum Sepolia to Starknet Sepolia. If you need testnet ETH on Ethereum Sepolia, you can use the [Chainlink faucet](https://faucets.chain.link/sepolia).
```

File: src/content/data-feeds/starknet/tutorials/snfoundry/consumer-contract.mdx [hunk 11]
```diff
@@ -0,0 +1,189 @@
...
+   ```bash
+   command: account deploy
+   transaction_hash: 0x541ffae49f77dd942376391286052d9fb87dc8d6848139ab1508c114e3aa005
+   ```
+
+   Wait a few seconds for the transaction to complete.
+
+### Deploy and interact with a consumer contract
+
+The consumer contract contains two functions that you'll interact with in this example:
```
Comment (dwightjl) [inline src/content/data-feeds/starknet/tutorials/snfoundry/consumer-contract.mdx:116]:
```suggestion
The consumer contract contains two functions that you interact with in this example:
```

**Review by dwightjl — APPROVED** [review]

**Pull request merged**